airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java

```diff
@@ -309,6 +310,29 @@ public void deleteDestinationDefinitionAndAssociations(final UUID destinationDef
     persistence.deleteConfig(definitionType, definitionId.toString());
   }
 
+  public void writeActorDefinitionWorkspaceGrant(final UUID actorDefinitionId, final UUID workspaceId) throws IOException {
+    database.query(ctx -> ctx.insertInto(ACTOR_DEFINITION_WORKSPACE_GRANT)
+        .set(ACTOR_DEFINITION_WORKSPACE_GRANT.ACTOR_DEFINITION_ID, actorDefinitionId)
+        .set(ACTOR_DEFINITION_WORKSPACE_GRANT.WORKSPACE_ID, workspaceId)
+        .execute());
+  }
+
+  public boolean actorDefinitionWorkspaceGrantExists(final UUID actorDefinitionId, final UUID workspaceId) throws IOException {
+    final Result<Record1<Integer>> records = database.query(ctx -> ctx.selectCount()
```

Comment:
This might be a little cleaner if you use `ctx.fetchCount` instead of `ctx.selectCount`, I think `fetchCount` will let you use an `int` return type directly instead of dealing with `Result<Record>` stuff

Reply:
ah cool, thanks for the tip!